bc_events/session.py

```diff
@@ -110,6 +110,32 @@ def publish(self, action=None, entity=None, data=None, category=None):
         topic = self.client.get_topic(category, entity, action)
         self._publish(topic, data)
 
+    def publish_all(self, events):
+        """Publish all events
+
+        Parameters
+        ----------
+        events : list
+            A list of events to publish
+        """
+
+        # If there are less than BULK_EVENT_SINGLE_POST_THRESHOLD events in the queue, publish individually
+        if len(self.events) <= BULK_EVENT_SINGLE_POST_THRESHOLD:
```

Comment:
I like the addition of a new function (currently `publish_all`) as a place to handle the `/events/bulk` endpoint. At the moment, `flush` is mostly just an alias for `publish_all`.

I think I would like to move this threshold logic into `flush` and leave `publish_all` (possibly renamed to `publish_bulk` or something similar?) to only handle the `/events/bulk` endpoint.